Better node install documentation (#283)
cloudflare/wrangler: README.md

### Using `npm`:

> 1. If you don't already have npm on your machine, install it using npm's recommended method: [a version manager](https://www.npmjs.com/get-npm).

Reviewer: ```suggestion
1. If you don't already have npm on your machine, install it using [npm's recommended method](https://www.npmjs.com/get-npm).
```

Author: This is definitely better, commiting.

>     If you have already installed npm with `brew` or `apt`, it is possible you will run into an `EACCES` error (related to how many system packagers install node) while installing wrangler. You can either uninstall that version of node and use the npm recommended install method, or use `cargo` instead.

Reviewer: this makes for weirdness; let's make this into a footnote after the two steps

Author: Will move the bit about system installers "inside" point #1 like the openssl one is.

> 1. Install Wrangler by running:

Reviewer: the numbering gets weird here; should be corrected somehow. likely would be corrected by making the bit about system package installers into a footnote

Author: Github still renders it fine, although the markdown numbers don't increment, the rendered output does. Will move the bit about system installers "inside" point #1 like the openssl one is.

Reviewer: i can't be the only one who opens readmes in my text editor locally

Author: Should be a separate PR as the entire doc is written with this numbering

### Using `npm`:

> 1. If you don't already have npm on your machine, install it using [npm's recommended method](https://www.npmjs.com/get-npm).

Reviewer 2: we should specify that this is using a node version manager and list/link some here. i think linking out to npm is good but i don't want to force people to click that link to get to where they need to go.

Reviewer 2: i think it's important to explain that an OS package manager will install node and it's global node_modules dir in a way that is only accessible to root. using the version manager, it will scope the node installation to the user and as a result will eliminate the need for root permissions

Author: I had this before: 

```
If you don't already have npm on your machine, install it using npm's recommended method: [a version manager](https://www.npmjs.com/get-npm).
```

And reverted it at @ashleymichal suggestion as I felt it was clearer without it, and the npm docs are fairly comprehensive wrt installing node.

Reviewer 2: the first thing you see when you scroll down the get-npm docs (which fwiw i wrote, many years ago) is this:

<img width="703" alt="Screen Shot 2019-06-28 at 3 35 25 PM" src="https://user-images.githubusercontent.com/1163554/60366732-68a32580-99ba-11e9-8183-6292f5a310f1.png">

we don't want folks to do that. we need to make it clear that the version manager strategy is the one to pick.

Author: ahhh I see. Will fix this

>     If you have already installed npm with `brew` or `apt`, it is possible you will run into an `EACCES` error (related to how many system packagers install node) while installing wrangler. You can either uninstall that version of node and use the npm recommended install method, or use `cargo` instead.

Reviewer 2: alternatively we can point people to the releases tab. they can manually download the binary for their machine and place it where they'd like

Author: I am concerned about recommending installing a raw binary that can't update itself

Reviewer 2: i hear you. i suggest it because its an option for people who are really stuck. in my workshop, there was one person who had to do it this way. since it exists i think it . is worth mentioning, as well as the building from source. 

we would call this a "manual install" and this is a common pattern for open source software.

Author: ```suggestion
    If you have already installed npm with a package manager, it is possible you will run into an `EACCES` error (related to how many system packagers install npm) while installing wrangler. You can either uninstall that version of npm and use the npm recommended install method (a version manager), use `cargo` instead, or download a binary from [our releases page ](https://github.com/cloudflare/wrangler/releases) and put it on your PATH.
```

Author: @ashleymichal Look good to you?

Reviewer: it's gotten way dense. it looks like it's worth calling out "manual install" in its own section after cargo. then all you have to say here is "you can either uninstall and reinstall as instructed above, or use an alternate wrangler installation method", call out nvm above (now i see why that is valuable, my bad for the earlier comment), and

>     If you have already installed npm with a package manager, it is possible you will run into an `EACCES` error (related to how many system packagers install npm) while installing wrangler. You can either uninstall npm and reinstall using the npm recommended install method (a version manager), or one of our other install methods.

Reviewer: ```suggestion
    If you have already installed npm with a package manager, it is possible you will run into an `EACCES` error while installing wrangler. This is related to how many system packagers install npm. You can either uninstall npm and reinstall using the npm recommended install method (a version manager), or one of our other install methods.
```

    ```
    $ brew install openssl@1.1
    ```

> ### Manual Install:

Reviewer: 👍
